modules/carbonRtdProvider.js

```diff
@@ -0,0 +1,242 @@
+  return matchedRules.map(x => x.Id);
+}
+
+export function prepareGPTTargeting(carbonData) {
+  if (isGptPubadsDefined()) {
+    setGPTTargeting(carbonData)
+  } else {
+    window.googletag = window.googletag || {};
+    window.googletag.cmd = window.googletag.cmd || [];
+    window.googletag.cmd.push(() => setGPTTargeting(carbonData));
+  }
+}
+
+export function setGPTTargeting(carbonData) {
+  if (Array.isArray(carbonData?.profile?.audiences)) {
+    window.googletag.pubads().setTargeting('carbon_segment', carbonData.profile.audiences);
+  }
+
+  if (Array.isArray(carbonData?.context?.pageContext?.contextualclassifications)) {
+    let contextSegments = carbonData.context.pageContext.contextualclassifications.map(x => {
+      if (x.type && x.type == 'iab_intent' && x.id) {
+        return x.id;
+      }
+    }).filter(x => x != undefined);
+    window.googletag.pubads().setTargeting('cc-iab-class-id', contextSegments);
+  }
+
+  if (Array.isArray(carbonData?.context?.customTaxonomy)) {
+    let customTaxonomyResults = matchCustomTaxonomy(carbonData.context.customTaxonomy);
+    window.googletag.pubads().setTargeting('cc-custom-taxonomy', customTaxonomyResults);
+  }
+}
+
+export function setPrebidConfig(carbonData) {
+  const ortbData = { user: {}, site: {} };
+
+  if (Array.isArray(carbonData?.profile?.audiences)) {
+    let userSegments = carbonData.profile.audiences.map(function (x) { return { id: x } });
+    ortbData.user.data = [{
+      name: 'www.ccgateway.net',
+      ext: { segtax: 507 }, // 507 Magnite Custom Audiences
+      segment: userSegments
+    }];
+  }
+
+  if (Array.isArray(carbonData?.context?.pageContext?.contextualclassifications)) {
+    let contextSegments = carbonData.context.pageContext.contextualclassifications.map(function (x) {
+      if (x.type && x.type == 'iab_intent' && x.id) {
+        return { id: x.id };
+      }
+    }).filter(x => x !== undefined);
+    ortbData.site.content = ortbData.site.content || {};
+    ortbData.site.content.data = [{
+      name: 'www.ccgateway.net',
+      ext: { segtax: 2 },
+      segment: contextSegments
+    }];
+  }
+
+  if (Array.isArray(carbonData?.context?.customTaxonomy)) {
+    ortbData.site.ext = ortbData.site.ext || {};
+    ortbData.site.ext.data = ortbData.site.ext.data || {};
+    ortbData.site.ext.data.customTaxonomy = matchCustomTaxonomy(carbonData.context.customTaxonomy);
+  }
+
+  if (Array.isArray(carbonData?.context?.dealIds)) {
+    ortbData.site.ext = ortbData.site.ext || {};
+    ortbData.site.ext.data = ortbData.site.ext.data || {};
+    ortbData.site.ext.data.dealIds = carbonData.context.dealIds;
+  }
+
+  sourceConfig.mergeConfig({ortb2: ortbData});
+}
+
+export function updateRealTimeDataAsync(callback, userConsent) {
+  let doc = window.top.document;
+  let pageUrl = `${doc.location.protocol}//${doc.location.host}${doc.location.pathname}`;
+  let profileId = storage.getDataFromLocalStorage(PROFILE_ID_KEY) || '';
```

Comment:
if carbon is blocked from storage this will result in empty string passed to RTD Carbon endpoint

Which the carbon endpoint will reject

Is this expected?

![image](https://user-images.githubusercontent.com/13972794/228322639-49b10971-8c3d-4a3e-aa10-418c9dd0fb67.png)

![image](https://user-images.githubusercontent.com/13972794/228322687-52759d86-f25e-40f8-98b4-483a6ee7ee59.png)

Reply:
If this is expected, then perhaps we should either

1. Log a warning to console and NOT send request to CARBON
2. Just make up a random profile ID

Not sure what the use case would dictate when carbon is denied storage access. Should we still be sending things?